pkg/types/aws_resource_manager.go

```diff
@@ -67,6 +68,13 @@ type AWSResourceManager interface {
 	// GetAttributes operations but all we have (for new CRs at least) is a
 	// name for the resource
 	ARNFromName(string) string
+	// LateInitialize returns an AWS Resource after setting the late initialized
+	// fields from the ReadOne call. This method will initialize the optional fields
+	// which were not provided by the k8s user but were defaulted by the AWS service.
+	// If there are no such fields to be initialized, the returned object is identical to
+	// object passed in the parameter.
+	// This method also adds/updates the ConditionTypeLateInitialized for the AWSResource.
+	LateInitialize(context.Context, AWSResource) (AWSResource, *requeue.RequeueNeededAfter)
```

Comment:
have this return `(AWSResource, error)` (`error` is an interface, implemented by `*requeue.RequeueNeededAfter`)